ludwig/backend/ray.py

```diff
@@ -1008,6 +1017,26 @@ def get_available_resources(self) -> Resources:
         resources = ray.cluster_resources()
         return Resources(cpus=resources.get("CPU", 0), gpus=resources.get("GPU", 0))
 
+    def max_concurrent_trials(self, hyperopt_config: Dict[str, Any]) -> Union[int, None]:
+        cpus_per_trial = hyperopt_config[EXECUTOR].get(CPU_RESOURCES_PER_TRIAL, 1)
+        num_cpus_available = self.get_available_resources().cpus
+
+        # No actors will compete for ray datasets tasks dataset tasks are cpu bound
+        if cpus_per_trial == 0:
+            return None
+
+        if num_cpus_available < 2:
+            raise RuntimeError("At least 2 CPUs are required for hyperopt when using a RayBackend.")
+
+        # Ray requires at least 2 free CPUs to ensure trials don't stall
+        max_possible_trials = int((num_cpus_available - 2) // cpus_per_trial)
```

Comment:
Are you sure we need 2 free CPUs? I believe it should just be 1. But please correct me if I'm missing something!